src/iface/interface.rs

```diff
@@ -2417,7 +2510,6 @@ impl<'a> InterfaceInner<'a> {
 
     fn dispatch_ip<Tx: TxToken>(&mut self, tx_token: Tx, packet: IpPacket) -> Result<()> {
         let ip_repr = packet.ip_repr();
-        assert!(!ip_repr.src_addr().is_unspecified());
```

Comment:
This also feels a little dicey, but as noted in the comment when crafting the mldv2 report packet, [RFC 3810](https://datatracker.ietf.org/doc/html/rfc3810#section-5.2.13) instructs that the source address can be set to the unspecified address:

> An MLDv2 Report MUST be sent with a valid IPv6 link-local source
   address, or the unspecified address (::), if the sending interface
   has not acquired a valid link-local address yet.  Sending reports
   with the unspecified address is allowed to support the use of IP
   multicast in the Neighbor Discovery Protocol [[RFC2461](https://datatracker.ietf.org/doc/html/rfc2461)].

Reply:
It seems #600 suggests to remove this assert as well?